pkg/utils/gardener/garden.go

```diff
@@ -292,18 +293,23 @@ func InjectGenericGardenKubeconfig(obj runtime.Object, genericKubeconfigName, ac
 		genericKubeconfigName,
 		accessSecretName,
 		"garden-kubeconfig",
-		VolumeMountPathGenericGardenKubeconfig,
+		mountPath,
 		containerNames...,
 	); err != nil {
 		return err
 	}
 
+	// inject reference annotations for generic kubeconfig
+	if err := references.InjectAnnotations(obj); err != nil {
+		return err
+	}
```

Comment:
Was this just missing before? Or why did you add it here?

Reply:
@timuthy

Reply:
@timuthy ?

Reply:
Makes sense, thank you!